pkg/pillar/cmd/zedagent/handlebaseos.go

```diff
@@ -78,74 +60,25 @@ func lookupZbootStatus(ctx *zedagentContext, key string) *types.ZbootStatus {
 	return &status
 }
 
-func publishZbootConfig(ctx *zedagentContext, config *types.ZbootConfig) {
-
-	key := config.Key()
-	log.Debugf("publishZbootConfig key %s TestComplete %v\n",
-		key, config.TestComplete)
-	pub := ctx.pubZbootConfig
-	pub.Publish(key, config)
-}
-
-// mark the zedcloud health/connectivity test complete flag
-// for baseosmgr to pick up and complete the partition activation
-func initiateBaseOsZedCloudTestComplete(ctx *zedagentContext) {
-
-	log.Infof("initiateBaseOsZedCloudTestComplete():\n")
-	partitionNames := []string{"IMGA", "IMGB"}
-	for _, key := range partitionNames {
-		config := lookupZbootConfig(ctx, key)
-		if config == nil {
-			config = &types.ZbootConfig{PartitionLabel: key}
-		}
-		if config.TestComplete {
-			continue
-		}
-		status := lookupZbootStatus(ctx, key)
-		if status != nil && status.PartitionLabel != "" {
-			if isBaseOsCurrentPartition(ctx, status.PartitionLabel) {
-				log.Infof("initiateBaseOsZedCloudTestComplete(%s): done\n", key)
-				config.TestComplete = true
-				publishZbootConfig(ctx, config)
-			}
-		}
-	}
-}
-
-func doZbootTestComplete(ctx *zedagentContext, status types.ZbootStatus) {
-	key := status.Key()
-	if !status.TestComplete {
-		log.Infof("doZbootTestComplete(%s): not TestComplete\n",
-			key)
-	} else {
-		log.Infof("doZbootTestComplete(%s):\n", key)
-		config := lookupZbootConfig(ctx, key)
-		if config == nil {
-			config = &types.ZbootConfig{PartitionLabel: key}
-		}
-		config.TestComplete = false
-		publishZbootConfig(ctx, config)
-		log.Infof("doZbootTestComplete(%s): done\n", key)
-
-		if ctx.rebootCmdDeferred {
-			log.Infof("TestComplete and deferred reboot\n")
-			ctx.rebootCmdDeferred = false
-			duration := time.Second * time.Duration(rebootDelay)
-			rebootTimer = time.NewTimer(duration)
-			go handleReboot(ctx.getconfigCtx)
-		}
+// on baseos install and activate set, the device reboot is initiated
+func initiateDeviceReboot(ctx *zedagentContext, infoStr string) {
```

Comment:
This pattern of a boolean to defer work is odd.
Can you change the ctx,deviceReboot to be a channel which you send on here, and the select statements receive on it? Such an approach is more maintainable IMHO.

Reply:
have moved the functionality to  a channel/reboot handler task primitive.